src/wallet/wallet.cpp

```diff
@@ -2018,7 +2040,17 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate)
                         sidechain.SafeguardMargin() - nHeight;
 
                     if (CSidechain::State::CEASED == pcoinsTip->isCeasedAtHeight(cert.GetScId(), chainActive.Height()))
-                        areBwtVoided = true;
+                    {
+                        LogPrint("cert", "%s():%d - cert[%s] is CEASED\n", __func__, __LINE__, cert.GetHash().ToString());
+                        areBwtStripped = true;
+                    }
+
+                    if (pcoinsTip->IsBwtStripped(cert.GetHash()) )
```

Comment:
`else if` ?

Reply:
I find `IsBwtStripped` not precise, since, for a given certificate, bwt may be not available because they are normally spent rather than voiding. A future dev may be tricked into thinking that `IsBwtStripped` is able to distinguish voided outputs from spent ones.
I would use instead a method like `HighQualityCertDataFor(block)` to inspect the current block and find all the low quality certs whose bwts must be voided (both in current block and in view)

Reply:
Hasty comment above. In given function wallet has been reconstructed from past blocks, while `pCoinsTip `view already refers to `chainActive.Tip()`. As a consequence, we are not fully able to reconstruct reason why cert bwts are voided (normally spent? last cert of ceased sc? low quality cert?).
Currently no best solution that flagging all scenario above with `allBwtsCleared `flag and move on